### DISCUSSION TITLE
3.1.4. TSML Legacy showing Updates, Not TSML UI

### DISCUSSION THREAD

**Discussion by Grateful-Viv** [code4recovery/12-step-meeting-list#773] [Q&A] [answered]
Trying to test out the differences between the two on staging site -- especially the way it works on mobile. I noticed  when viewing a meeting on mobile it said "Updated Invalid Date". So I reimported CSV data (delete all meetings, locations, groups, districts, and regions). It reports the exact number of meetings correctly in the dashboard. It looks fine in Legacy. However, TSML UI shows all the old meeting data.  I can toggle back and forth between legacy (which shows fine) and UI (which shows all old data).  Purging cache makes no difference. And it still shows on mobile "Updated Invalid Date". 

Is there a lag time between an actual update and what is displayed in UI?

**Comment by Grateful-Viv** [comment]
UPDATE: It shows on both my staging sites the message "Updated Invalid Date" on mobile. The other staging site I just looked at had good data from a week ago and I made no changes to it other than to install the TSML 3.14.1 release.

**Reply by Grateful-Viv** [reply]
It also shows “Updated Invalid Date” on my iPad. Screen shot below is from live site using the shortcode. 

![A4FEC6BB-AF7D-48D4-94A1-9806149730BD](https://user-images.githubusercontent.com/62247421/163687667-88e2498f-52bd-4e27-8db4-45d577f90ef9.png)

**Reply by Grateful-Viv** [reply]
It’s also showing up same way on your demo site. 

![7AF5F13E-9232-4A8F-A01F-71BCF8701574](https://user-images.githubusercontent.com/62247421/163687798-3ea62863-c4e4-49fd-b8eb-78bdc5d06361.png)

**Reply by kiyote33** [reply]
Hi @Grateful-Viv, it's interesting that you're getting that on our test site! I've checked half a dozen meetings on demo.code4recovery.org and they all display the date correctly.  I've also tried to reproduce in my sandbox, to no avail.  I'm wondering if there might be some local date format setting in your environment that's come into play here.  Could you export your meetings that are generating the invalid date to a csv file and post it here?  I can then verify that it displays correctly or not in my sandbox.  Thx...

**Reply by Grateful-Viv** [reply]
I don't mind sharing my data with you, but it is very curious why it also shows up on my  iOS devices (15.4.1) for phone and iPad with **Update Invalid Date** on your demo site too. 

It looks fine on the Windows desktop.

I will add that it doesn't appear this way looking at https://aasanjose.org/meetings with my iPhone or iPad.

Also, there is definitely a lag time between doing updates and having it appear on the site. I just updated a meeting to see if today's date would make any difference. Even though the date is updated, it does not show up for a long while, but it does show up immediately on the legacy UI.

I downloaded the CSV and it shows correctly the date and time for the meeting I just updated. It's the meeting on the last row. Please tell me where to send the CSV if you want to see it.

**Reply by kiyote33** [reply]
You can send it to waynep@code4recovery.org.  Thx...

**Comment by joshreisner** [comment, marked as answer]
interesting! i [created an issue](https://github.com/code4recovery/tsml-ui/issues/224) for the updated date problem

re the time lag, your browser tends to cache the JSON file. this will be cleared when you quit and reopen the browser, or if you do a hard refresh. in chrome i believe this is command shift r. it can differ between browsers.

there are cache-busting techniques that TSML could implement, like appending the last modified date of the query string of the cache address, eg `/wp-content/tsml-cache-238f7s7.json?79823479238749249`

**Reply by Grateful-Viv** [reply]
Want to add the Update Invalid Date also appears on my iPhone using both Chrome & Firefox apps, not just Safari.

**Reply by Grateful-Viv** [reply]
`re the time lag, your browser tends to cache the JSON file. this will be cleared when you quit and reopen the browser, or if you do a hard refresh. in chrome i believe this is command shift r. it can differ between browsers.`

@joshreisner, I did a lot of that yesterday. In Windows, it's CTRL-R or F5. I also closed my browser. I also flushed my browser cache using Chrome tools. I could try this on Firefox, but I saw the delay in seeing updates on my iOS devices too. The legacy UI showed an instantaneous update, no matter what browser I was using.
